core/src/main/java/hudson/util/ProcessTree.java

```diff
@@ -241,13 +241,22 @@ void killByKiller() throws InterruptedException {
          * null if no one objects killing the process.
          */
         protected @CheckForNull VetoCause getVeto() {
-            for (ProcessKillingVeto vetoExtension : ProcessKillingVeto.all()) {
-                VetoCause cause = vetoExtension.vetoProcessKilling(this);
-                if (cause != null) {
-                    if (LOGGER.isLoggable(FINEST))
-                        LOGGER.finest("Killing of pid " + getPid() + " vetoed by " + vetoExtension.getClass().getName() + ": " + cause.getMessage());
-                    return cause;
+            String causeMessage = null;
+            
+            try {
+                VirtualChannel channelToMaster = SlaveComputer.getChannelToMaster();
```

Comment:
IIUC it won't work properly on the master, because there the result will be null